lib/vector-common/src/constants.rs

```diff
@@ -0,0 +1,2 @@
+pub const GZIP_MAGIC: &[u8] = &[0x1f, 0x8b];
```

Comment:
@pront  Moved shared magic bytes to here.

There is a few places where compression detection is repeated. For example `is_gzipped` func from `lib/file-source/src/file_watcher/mod.rs`, `is_gzip` func from `src/sources/aws_kinesis_firehose/handlers.rs` and inside the `from_magic` method from `lib/codecs/src/decoding/framing/chunked_gelf.rs`

Maybe it would be worth to refactor all of those detections (and also handle flate2 decompression with shared code) into a `vector-common` `compression` module as done with this `constants` module. But I think it should be done in a follow-up PR. (and there will be likely more places where this detection is done).

This seems like a good start, at least for the scope of this PR though.

Reply:
Thank you, this is a good first step. Agree it's best to follow-up in another PR with the other changes you described here.